packages/compiler/src/experimental/state-accessor.ts

```diff
@@ -0,0 +1,30 @@
+import type { Program } from "../core/program.js";
+import type { Type } from "../core/types.js";
+
+type StateMapGetter<K extends Type, V> = (program: Program, type: K) => V | undefined;
+type StateMapSetter<K extends Type, V> = (program: Program, type: K, value: V) => void;
+type StateMapMapGetter<K extends Type, V> = (program: Program) => Map<K, V>;
+
+/** @deprecated use `useStateMap` from `@typespec/compiler/utils` instead */
+export function unsafe_useStateMap<K extends Type, V>(
```

Comment:
does doing `export const unsafe_useStateMap = useStateMap;`

not work so it doesn't have to be redefined.

Reply:
In order for the `@deprecated` tag to be recognized, it has to be something [like this](https://stackoverflow.com/a/76057689):
```typescript
import { useStateMap } from "../utils/state-accessor.js";
/** @deprecated use `useStateMap` from `@typespec/compiler/utils` instead */
export const unsafe_useStateMap = useStateMap;
```

If that's preferable, I can make the change. I thought there might be some utility in redefining, so that further changes to the "real" functions don't affect the deprecated usages.